src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/index.js

```diff
@@ -108,18 +119,62 @@ const DesignColorsPanel = forwardRef(
 		const [ selectedPalette, setSelectedPalette ] = useState( null );
 		const [ colorPickerCalledBy, setColorPickerCalledBy ] = useState( '' );
 		const [ showColorPicker, setShowColorPicker ] = useState( false );
-		const customPaletteId = colors.length - 1;
+
+		useEffect( () => {
+			const slug = currentData.sitegen?.homepages?.active?.slug;
+			if ( slug && ! customColors ) {
+				const storedCustomColors =
+					currentData.sitegen.homepages.data[ slug ].color
+						.customColors;
+				if ( storedCustomColors ) {
+					setCustomColors( storedCustomColors );
+				} else {
+					const defaultPalette =
+						currentData.sitegen.homepages.data[ slug ].color
+							.palette;
+					const defaultCustomColors = {
+						primary: '',
+						secondary: '',
+						tertiary: '',
+					};
+					for ( const item of defaultPalette ) {
+						switch ( item.slug ) {
+							case 'primary':
+								defaultCustomColors.primary = item.color;
+								break;
+							case 'secondary':
+								defaultCustomColors.secondary = item.color;
+								break;
+							case 'tertiary':
+								defaultCustomColors.tertiary = item.color;
+								break;
+							default:
+								break;
+						}
+					}
+
+					if ( ! defaultCustomColors.secondary ) {
+						defaultCustomColors.secondary = defaultPalette.find(
+							( item ) => item.slug === 'base'
+						).color;
+					}
+
+					setCustomColors( defaultCustomColors );
+				}
+			}
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+		}, [ currentData ] );
 
 		const handleApplyCustomColors = () => {
 			setSelectedCustomColors( true );
 			setIsEditingCustomColors( false );
-			setSelectedPalette( customPaletteId );
-			colors[ selectedPalette ] = selectedColor;
+			setSelectedPalette( 'custom' );
+			setCustomColors( selectedColor );
```

Comment:
We should try our best to use as few states as possible. Using the `slug` as custom in the palette itself will allow us to combine the extra custom colors state into selected colors?